File: fixtures/canary-python/deprecation_canary.py
# validate-provider canary for kanarienkrebs' python-dev lane.
# Plain `python3 deprecation_canary.py`                 -> warns, exits 0.
# With PYTHONWARNINGS=error::DeprecationWarning (dev layer) -> the warning is
# raised as an exception, exits nonzero.
# The lane is proven live only if the dev layer flips 0 -> nonzero.
import warnings

warnings.warn(
    "kanarienkrebs canary — this deprecation MUST become an error under the dev layer",
    DeprecationWarning,
)
print("canary survived: if you see this WITH the layer, the layer is NOT active")

File: fixtures/canary-runtime/deprecation-canary.mjs
// validate-provider canary for the kanarienkrebs ts-runtime lane.
// Plain `node canary.mjs`               -> emits a warning, exits 0.
// `node --throw-deprecation canary.mjs` -> the warning THROWS, exits nonzero.
// The lane is proven live only if the strict layer flips 0 -> nonzero.
process.emitWarning("kanarienkrebs canary — this deprecation MUST throw under the layer", {
  type: "DeprecationWarning",
});
console.log("canary survived: if you see this WITH the layer, the layer is NOT active");
